architecture/adr-list/adr-001-cheqd-did-method.md

```diff
@@ -0,0 +1,616 @@
+
+
+#### Example of DIDDoc representation
+
+
+```jsonc
+{
+  "@context": [
+    "https://www.w3.org/ns/did/v1",
+    "https://w3id.org/security/suites/ed25519-2020/v1"
+  ],
+  "id": "did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612",
+  "verificationMethod": [
+    {
+      "id": "did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612#authKey1",
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612",
+      "publicKeyMultibase": "zAKJP3f7BD6W4iWEQ9jwndVTCBq8ua2Utt8EEjJ6Vxsf"
+    },
+    {
+      "id": "did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612#capabilityInvocationKey",
+      "type": "Ed25519VerificationKey2020", // external (property value)
+      "controller": "did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612",
+      "publicKeyMultibase": "z4BWwfeqdp1obQptLLMvPNgBw48p7og1ie6Hf9p5nTpNN"
+    }
+  ],
+  "authentication": ["did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612#authKey1"],
+  "capabilityInvocation": ["did:cheqd:mainnet:c82f2b02-bdab-4dd7-b833-3e143745d612#capabilityInvocationKey"],
+}
+```
+
+#### State format for DIDDocs on ledger
+
+`"diddoc:<id>" -> {DIDDoc, DidDocumentMetadata, txHash, txTimestamp }`
+
+`didDocumentMetadata` is created by the node after transaction ordering and before adding it to a state.
+
+### DID Document metadata
+
+Each DID Document MUST have a metadata section when a representation is produced. It can have the following properties:
+
+1. **`created`** (string): Formatted as an XML Datetime normalized to UTC 00:00:00 and without sub-second decimal precision, e.g., `2020-12-20T19:17:47Z`.
+2. **`updated`** (string): The value of the property MUST follow the same
+formatting rules as the created property. The `updated` field is `null` if an Update operation has never been performed on the DID document. If an updated property exists, it can be the same value as the created property when the difference between the two timestamps is less than one second.
+3. **`deactivated`** (string): If DID has been deactivated, DID document metadata MUST include this property with the boolean value `true`. By default this is set to `false`.
+4. **`versionId`** (string): A UUID string that represents the version identifier of the DID Document.
+5. **`resources`** (list of resources metadata referred to as [Resource previews](adr-002-did-linked-resources.md))| *optional*). Cannot be changed by CreateDID or UpdateDID transactions. cheqd ledger stores only the resource identifiers in the DID Doc metadata. The remainder of the resources' metadata is added when a DID is resolved.
+6. **`previousVersionId`** (string): A UUID string that represents the version identifier of the previous version of the DID Document. The `previousVersionId` field is `null` if an Update operation has never been performed on the DID document
```

Comment:
`previousVersionId` should be `""` (an empty string) instead of null because currently resolver returns in this format